plugins/com.google.cloud.tools.eclipse.appengine.libraries.ui/src/com/google/cloud/tools/eclipse/appengine/libraries/ui/CloudLibrariesPage.java

```diff
@@ -147,16 +153,22 @@ public boolean finish() {
         /*
          * FIXME: BuildPath.addNativeLibrary() is too heavy-weight here. ClasspathContainerWizard,
          * our wizard, is responsible for installing the classpath entry returned by getSelection(),
-         * which will perform the library resolution. We just need to save the selected libraries 
+         * which will perform the library resolution. We just need to save the selected libraries
          * so that they are resolved later.
          */
+        BuildPath.saveLibraryList(project, libraries, new NullProgressMonitor());
         Library masterLibrary =
             BuildPath.collectLibraryFiles(project, libraries, new NullProgressMonitor());
         newEntry = BuildPath.computeEntry(project, masterLibrary, new NullProgressMonitor());
-        BuildPath.saveLibraryList(project, libraries, new NullProgressMonitor());
         if (newEntry == null) {
-          // container-editing only refreshes the content if new
-          BuildPath.runContainerResolverJob(project);
+          // existing entry needs to be updated
+          ClasspathContainerInitializer initializer =
+              JavaCore.getClasspathContainerInitializer(
+                  LibraryClasspathContainer.CONTAINER_PATH_PREFIX);
+          if (initializer.canUpdateClasspathContainer(originalEntry.getPath(), project)) {
```

Comment:
Can you add a comment like "this is always true for our classpath container"? I was spending some time wondering what will happen if this say "you can't update, so you can't add libraries" while there is no feedback to user that the libraries have not been added.